Mathlib/Data/ZMod/Basic.lean

```diff
@@ -81,6 +80,11 @@ theorem val_mul' {m n : ZMod 0} : (m * n).val = m.val * n.val :=
   Int.natAbs_mul m n
 #align zmod.val_mul' ZMod.val_mul'
 
+lemma eq_one_of_isUnit_natCast {n : ℕ} (h : IsUnit (n : ZMod 0)) : n = 1 := by
+  have := Int.isUnit_iff.mp h
+  norm_cast at this
+  exact this.resolve_right not_false
+
```

Comment:
You can easily generalize this to `IsUnit n <-> n.val = 1` for `n : ZMod 0`, and make use of `val_nat_cast` below:
```lean
theorem val_unit' {n : ZMod 0} : IsUnit n ↔ n.val = 1 := by
  simp only [val]
  rw [Int.isUnit_iff, Int.natAbs_eq_iff, Nat.cast_one]

lemma eq_one_of_isUnit_natCast {n : ℕ} (h : IsUnit (n : ZMod 0)) : n = 1 := by
  rw [← Nat.mod_zero n, ← val_nat_cast, val_unit'.mp h]
```